api/src/scripts/translateResourceDescriptions.js

```diff
@@ -0,0 +1,126 @@
+ * Language can be any of "es" (Spanish), "fr" (French), or "zh" (Simplified Chinese)
+ */
+
+const dotenv = require('dotenv');
+const mongoose = require('mongoose');
+const path = require('path');
+const Resource = require('../models/resource');
+const Translation = require('../models/translation');
+
+// Connecting to our *dev* database - we can change this later once we're confident the script works
+dotenv.config({
+  path: path.resolve(__dirname, `../../config/dev.env`),
+});
+
+mongoose.connect(process.env.MONGO_URL, {
+  useNewUrlParser: true,
+  useUnifiedTopology: true,
+});
+
+mongoose.Promise = global.Promise;
+
+mongoose.connection
+  .once('open', () => console.log('Connected to MongoDB instance.'))
+  .on('error', (error) => console.log('Error connecting to MongoDB:', error));
+
+/**
+ * Step 1: Get the descriptions of every Resource in our database
+ */
+// Change this to get just the descriptions and their associated resource `_id` - so we have an array of objects with fields _id and description
+const resourceDescriptions = Resource.find({}, { _id: 1, description: 1 });
+
+/**
+ * Step 2: Call the Google API to translate those descriptions to the target language
+ *
+ * Essentially we want to feed in our array from step 1 into the google API, and save an array of objects with fields _id and translated text
+ *
+ * https://cloud.google.com/translate/docs/reference/rest/v2/translate - this is the POST request we want
+ * We can make the POST request using `fetch` - you can look in client/src/utils/auth.js for examples
+ * Make sure to specify the `source` as "en"
+ * https://stackoverflow.com/questions/35139572/sample-code-for-google-translate-api - this may help on how to format the request URL
+ *
+ * For now, you can go ahead and write this code, but I'll supply the API key later (basically, I don't want to waste API credits) - to test Step 3,
+ * you can mock out the resultant data to feed into Step 3
+ */
+const apiKey = '';
+// map ids of resources to translated descriptions
+var translatedResourceDescriptions = new Map();
+const languageType = process.argv.slice(2)[0]; // the language specified in the argument when running the script - the "target" argument in the POST request
+resourceDescriptions.forEach(function (resource) {
+  var id = resource._id;
+  var descriptionText = resource.description;
+  var source =
+    `https://www.googleapis.com/language/translate/v2?key=${apiKey}&source=en&target=${language}&callback=translateText&q=` +
+    descriptionText;
+  async (body) => {
+    const { resourceDescriptions } = body;
+    const res = await fetch(source, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        resourceDescriptions,
+      }),
+    });
+    return translatedResourceDescriptions.set(
+      id,
+      res.json().body.data.translations[0].translatedText, // store translated descriptions with id
+    );
+  };
```

Comment:
I'm a bit confused with the code here - I'm unsure if this needs to be wrapped in an `async() => {}` (but lmk if it does).  I would imagine you can just do something like this instead of the lines I highlighted:
```
fetch(source, {
  method: 'POST',
  headers: {
    'Content-Type': 'application/json',
  }
}).then((res) =>
  translatedResourceDescriptions.set(id, res.json().body.data.translations[0].translatedText)
);
```
We don't need a body since all of our "parameters" are in the source :)
Hopefully this makes it a bit simpler too

Reply:
Thanks Jackie!